lib/core/base/metadata-function-map.js

```diff
@@ -0,0 +1,9 @@
+/*eslint no-unused-vars:0*/
+// TODO: es-modules-checks
+// TODO: es-modules-rules
+// import all check evaluate and after functions, and all rule matches functions
+
+const metadataFunctionMap = new Map();
```

Comment:
I thought we were going to have a map for checks and a map for rules? How will we differenciate between the different types of functions here?

For example:

```
map.set('some-fn', (some, args) => { /* ... */ })
```

Is the above for a check `eval`, a check `after` or a rule `matches`?

Reply:
The id for each function will be unique and include the type in the name. E.g. `aria-allowed-attr-matches`, `aria-allowed-attr-evaluate`, `aria-allowed-attr-after`